server/controllers/auth.js

```diff
@@ -0,0 +1,48 @@
+const User = require('../models/user');
+const HttpStatus = require('http-status-codes');
+const passport = require('passport');
+
+// register controller
+exports.SignUpController = async (req, res, next) => {
+    try {
+        const { email, password , firstName, lastName } = req.body;
+        const userExists = await User.exists({ email });
+        if (userExists) {
+            res.status(HttpStatus.CONFLICT).send({ response: `Email ${email} already used!` });
+            return;
+        }
+        const user = new User({
+            email, password, firstName, lastName
+        });
+        await user.save();
+        //authenticate the user after the registration
+        Authenticate(req, res, next, HttpStatus.CREATED)
+    } catch (error) {
+        next(error)
+    }
+}
+//login controller
+exports.SignInController = async (req, res, next) => {
+    Authenticate(req, res, next, HttpStatus.OK)
+}
+
+function Authenticate(req, res, next, statusCode) {
+    passport.authenticate('local', (err, user) => {
+      if (err) { next(err); }
+      if (!user) {
+        return res.status(HttpStatus.FORBIDDEN).send(user);
+      }
+      req.logIn(user, async (loginErr) => {
```

Comment:
does req.logIn come from passport-local and it'll add the set-cookie header in the API request?

Reply:
Yes, in fact Passport exposes a **login()** function on **req** which can be used to establish a login session (this is described in the official Passsport.js documentation).
I used the **LocalStrategy** (passport-local) strategy to establish authentication by username / password and he is the responsible for setting the token in the cookies.